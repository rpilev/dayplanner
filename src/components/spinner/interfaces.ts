export interface Spinner {
  speed?: number
}
