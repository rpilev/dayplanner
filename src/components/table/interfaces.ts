import { ReactNode } from 'react'

export interface Table {
  children: ReactNode
  isLoading?: boolean
}

export interface Td {
  center?: boolean
  fixedWidth?: number
  oneLine?: boolean
  multiLine?: boolean
}

export interface Th {
  tiny?: boolean
}
