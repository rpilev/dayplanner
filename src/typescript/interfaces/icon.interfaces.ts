export interface Icon {
  name: string
  size: string
}

export interface Sizes {
  [key: string]: number
}

export interface IconWrapper {
  size: string
}

export interface SVGS {
  [key: string]: string
}
