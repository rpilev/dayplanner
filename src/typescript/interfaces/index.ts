export * as TaskInterfaces from './task.interfaces'
export * as IconInterfaces from './icon.interfaces'
