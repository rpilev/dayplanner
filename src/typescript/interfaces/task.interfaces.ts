export interface Task {
  id: number
  name: string
  description: string
  done: boolean
}

export interface TasksState {
  list: Task[]
}

export interface RootState {
  tasks: TasksState
}

export interface Td {
  center?: boolean
  fixedWidth?: number
  oneLine?: boolean
  multiLine?: boolean
}

export interface Th {
  tiny?: boolean
}

export interface Table {
  tasks: Task[]
}

export interface GetTasksResponse {
  content: [Task]
}
