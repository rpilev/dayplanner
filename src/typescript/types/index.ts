export * as Types from './task.types'
