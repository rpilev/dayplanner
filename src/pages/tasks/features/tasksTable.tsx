import React from 'react'
import TruncateMarkup from 'react-truncate-markup'

import { TaskInterfaces } from '@app/typescript/interfaces'
import { Table, Icon } from '@app/components'

export const TasksTable = ({ tasks }: TaskInterfaces.Table): JSX.Element => (
  <Table.Table>
    <thead>
      <tr>
        <Table.Th>Name:</Table.Th>
        <Table.Th>Description</Table.Th>
        <Table.Th tiny>Done</Table.Th>
      </tr>
    </thead>
    <tbody>
      {tasks.map(({ name, description, done, id }) => (
        <tr key={id}>
          <Table.Td fixedWidth={100}>{name}</Table.Td>
          <Table.Td multiLine>
            <TruncateMarkup lines={3}>
              <p>{description}</p>
            </TruncateMarkup>
          </Table.Td>
          <Table.Td center>{done && <Icon.Icon name="green-check" size="sm" />}</Table.Td>
        </tr>
      ))}
    </tbody>
  </Table.Table>
)
