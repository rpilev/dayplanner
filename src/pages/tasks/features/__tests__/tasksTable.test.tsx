import React from 'react'
import { render } from '@testing-library/react'

import { TaskInterfaces } from '@app/typescript/interfaces'

import { TasksTable } from '../tasksTable'

let tasks: TaskInterfaces.Task[]

beforeEach(() => {
  tasks = [
    {
      id: 1,
      name: 'name value 1',
      description: 'description value 1',
      done: true,
    },
    {
      id: 2,
      name: 'name value 2',
      description: 'description value 2',
      done: true,
    },
    {
      id: 3,
      name: 'name value 3',
      description: 'description value 3',
      done: false,
    },
  ]
})

describe('Table', () => {
  test('should render correctly', () => {
    const IconRender = render(<TasksTable tasks={tasks} />)

    expect(IconRender).toBeDefined()
    expect(IconRender).toMatchSnapshot()
  })
})
