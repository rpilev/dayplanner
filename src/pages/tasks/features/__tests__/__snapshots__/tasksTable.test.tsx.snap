// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="sc-bdnylx geRHdX"
      >
        <thead>
          <tr>
            <th
              class="sc-gtssRu lbqNjK"
            >
              Name:
            </th>
            <th
              class="sc-gtssRu lbqNjK"
            >
              Description
            </th>
            <th
              class="sc-gtssRu cZWRLh"
            >
              Done
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="sc-dlnjPT jKsEFD"
            >
              name value 1
            </td>
            <td
              class="sc-dlnjPT gFntqo"
            >
              <p
                style="word-wrap: break-word; display: block;"
              >
                d
                ...
              </p>
            </td>
            <td
              class="sc-dlnjPT cdTWoW"
            >
              <div
                class="sc-hKFyIo dpjfMn"
              />
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlnjPT jKsEFD"
            >
              name value 2
            </td>
            <td
              class="sc-dlnjPT gFntqo"
            >
              <p
                style="word-wrap: break-word; display: block;"
              >
                d
                ...
              </p>
            </td>
            <td
              class="sc-dlnjPT cdTWoW"
            >
              <div
                class="sc-hKFyIo dpjfMn"
              />
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlnjPT jKsEFD"
            >
              name value 3
            </td>
            <td
              class="sc-dlnjPT gFntqo"
            >
              <p
                style="word-wrap: break-word; display: block;"
              >
                d
                ...
              </p>
            </td>
            <td
              class="sc-dlnjPT cdTWoW"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="sc-bdnylx geRHdX"
    >
      <thead>
        <tr>
          <th
            class="sc-gtssRu lbqNjK"
          >
            Name:
          </th>
          <th
            class="sc-gtssRu lbqNjK"
          >
            Description
          </th>
          <th
            class="sc-gtssRu cZWRLh"
          >
            Done
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="sc-dlnjPT jKsEFD"
          >
            name value 1
          </td>
          <td
            class="sc-dlnjPT gFntqo"
          >
            <p
              style="word-wrap: break-word; display: block;"
            >
              d
              ...
            </p>
          </td>
          <td
            class="sc-dlnjPT cdTWoW"
          >
            <div
              class="sc-hKFyIo dpjfMn"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlnjPT jKsEFD"
          >
            name value 2
          </td>
          <td
            class="sc-dlnjPT gFntqo"
          >
            <p
              style="word-wrap: break-word; display: block;"
            >
              d
              ...
            </p>
          </td>
          <td
            class="sc-dlnjPT cdTWoW"
          >
            <div
              class="sc-hKFyIo dpjfMn"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlnjPT jKsEFD"
          >
            name value 3
          </td>
          <td
            class="sc-dlnjPT gFntqo"
          >
            <p
              style="word-wrap: break-word; display: block;"
            >
              d
              ...
            </p>
          </td>
          <td
            class="sc-dlnjPT cdTWoW"
          />
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
